### DISCUSSION TITLE
select in folder/file name issue

### DISCUSSION THREAD

**Discussion by jolabrenz** [UtrechtUniversity/yoda#182] [Q&A] [answered]
I have a question regarding this issue https://github.com/UtrechtUniversity/yoda/issues/102 

I noticed it is possible to imkdir a folder containing "select" and put files in with iput (irsync also creates a folder and puts files inq).  I wanted to delete them (files and folder). But I am failing to do so because irm is failing because of the issue with select. (iadmin rmdir and irmdir are failing as well).
 
Is there an easy way to remove the unwanted folder and content? Do I have to deactivate some of the YODA rules temporarily? 

Best regards
Johannes

**Comment by stsnel** [comment]
With which Yoda version have you encountered this issue? 

Do you see a Python exception in the rodsLog when you try to remove the collection? If so, can you share it?

Best regards,
Sietse

**Reply by jolabrenz** [reply]
Hi the YODA Version is 1.7

The Python exception:

Dec  1 19:32:24 pid:22358 remote addresses: 172.27.0.152, 172.27.1.13 ERROR: msiMakeGenQuery: fillGenQueryInpFromStrCond failed.
Dec  1 19:32:24 pid:22358 remote addresses: 172.27.0.152, 172.27.1.13 ERROR: caught python exception: Traceback (most recent call last):
  File "/etc/irods/rules_uu/util/rule.py", line 91, in r
    result = f(Context(callback, rei), *a)
  File "/etc/irods/rules_uu/util/policy.py", line 80, in r
    result = f(ctx, *args)
  File "/etc/irods/rules_uu/policies.py", line 163, in py_acPreprocForRmColl
    str(session_vars.get_map(ctx.rei)['collection']['name']))
  File "/etc/irods/rules_uu/policies.py", line 56, in can_coll_delete
    if folder.has_locks(ctx, coll) and not user.is_admin(ctx, actor):
  File "/etc/irods/rules_uu/folder.py", line 466, in has_locks
    return len(get_locks(ctx, coll, org_metadata=org_metadata)) > 0
  File "/etc/irods/rules_uu/folder.py", line 443, in get_locks
    org_metadata = get_org_metadata(ctx, path, object_type=object_type)
  File "/etc/irods/rules_uu/folder.py", line 437, in get_org_metadata
    else " AND COLL_NAME = '{}'".format(path)))]
  File "/etc/irods/rules_uu/util/query.py", line 202, in __iter__
    self.exec_if_not_yet_execed()
  File "/etc/irods/rules_uu/util/query.py", line 150, in exec_if_not_yet_execed
    irods_types.GenQueryInp())['arguments'][2]
RuntimeError: [iRods__Error__Code:-1107000] [-]	/irods/server/re/include/irods_re_plugin.hpp:325:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::rule_execution_manager_pack::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::rule_execution_manager_pack::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, RuleExecInfo *&, irods::unpack &&)>, As = <const std::__1::basic_string<char> &, RuleExecInfo *&, irods::unpack>] :  status [NO_COLUMN_NAME_FOUND]  errno [] -- message [exec_microservice_adapter failed]
	[-]	/irods/server/re/src/irods_re_plugin.cpp:132:irods::error irods::default_microservice_manager<RuleExecInfo *>::exec_microservice_adapter(std::string, default_ms_ctx, std::list<boost::any> &) :  status [NO_COLUMN_NAME_FOUND]  errno [] -- message [exec_microservice_adapter failed]

it is followed by the irods errors:

Dec  1 19:32:24 pid:22358 remote addresses: 172.27.0.152, 172.27.1.13 ERROR: executeRuleAction Failed for py_acPreprocForRmColl status = -1107000 NO_COLUMN_NAME_FOUND
Dec  1 19:32:24 pid:22358 DEBUG: 
Dec  1 19:32:24 pid:22358 remote addresses: 172.27.0.152, 172.27.1.13 ERROR: applyRule: -1089000, [-]	/irods/server/re/include/irods_re_plugin.hpp:325:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::rule_execution_manager_pack::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::rule_execution_manager_pack::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &)>, As = <const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &>] :  status [CUT_ACTION_PROCESSED_ERR]  errno [] -- message [applyRuleUpdateParams failed for rule acPreprocForRmColl]
	[-]	/irods/plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp:348:irods::error exec_rule(irods::default_re_ctx &, const std::string &, std::list<boost::any> &, irods::callback) :  status [CUT_ACTION_PROCESSED_ERR]  errno [] -- message [applyRuleUpdateParams failed for rule acPreprocForRmColl]


Best regards,
Johannes

**Comment by stsnel** [comment, marked as answer]
Hi Johannes,

We have a workaround for this particular issue. You can get it by updating the Yoda environment to the latest 1.7.x version using the Ansible playbook, or alternatively by applying/cherry-picking this commit on the ruleset of the provider: https://github.com/UtrechtUniversity/yoda-ruleset/commit/ffb8b133c0d6dbcaea5eab288059f23b35c9656a . After that, you should be able to remove the folders using a rodsadmin account.

Although iRODS 4.2.11 still has some GenQuery parser issues, all the issues related to "select" in names appear to be resolved, so upgrading to Yoda 1.8.x. would also resolve this particular issue for non-rodsadmin users.

Best regards,
Sietse

**Reply by jolabrenz** [reply]
Thanks a lot.
